internal/app/storage/db.go

```diff
@@ -0,0 +1,117 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	_ "github.com/jackc/pgx/v5/stdlib"
+	"github.com/shilin-anton/urlreducer/internal/app/model"
+	"strconv"
+)
+
+type DataBase struct {
+	DB *sql.DB
+}
+
+type DataBaseInterface interface {
+	Ping() error
+	AddRecord(short, url string) error
+	GetRecord(short string) (string, error)
+	AddBatch(model.BatchUrls) error
+}
+
+func NewDB(dataBaseDSN string) (*DataBase, error) {
+	db, err := sql.Open("pgx", dataBaseDSN)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = db.ExecContext(context.Background(), `CREATE TABLE IF NOT EXISTS urls (
+			id VARCHAR(50) NOT NULL,
+			url VARCHAR(255) NOT NULL UNIQUE,
+			short VARCHAR(50) NOT NULL
+		)`)
+	if err != nil {
+		db.Close()
+		return nil, err
+	}
+
+	return &DataBase{DB: db}, err
+}
+
+func (db *DataBase) AddRecord(short, url string) error {
```

Comment:
```suggestion
func (db *DataBase) AddRecord(ctx context.Context, short, url string) error {
```
Контекст здесь и везде в функциях, обрабатывающих пользовательские запросы, необходимо брать из контекста http-запроса (`r.Context()`), а не из воздyха. Соответственно, это должно отразиться и в интерфейсе.